### PR TITLE
enh(CI): Rework Jest RTL tests to Cypress Component testing for Local Authentication

### PR DESCRIPTION
**Pull request by SoldierBoy31**
## Description

Rework Jest RTL tests to Cypress Component testing for Local Authentication:

- Adding new LocalAuthentication spec file for cypress component testing.
- Updating commands.tsx on cypress, generalize response type for interceptRequestAPI function.
- Deleting of the index.test file.

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [X] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Run GHA CI/CD on github or run pnpm cypress component testing on local.

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+          ...defaultPasswordSecurityPolicyAPI.password_security_policy,
+          password_min_length: 45
+        }
+      }
+    });
+
+    cy.waitForRequest('@getMinLengthPasswordSecurityPolicyFromAPI');
+  });
+
+  it('updates the retrieved form recommended values and reset the form to the inital values', () => {
```
Comment (bdauria) [inline centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:141]:
```suggestion
  it('updates the retrieved form recommended values and reset the form to the inital values when the "Reset" button is clicked', () => {
```

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+        .eq(0)
+        .should('be.visible');
+      cy.findAllByLabelText(labelPasswordMustContainSpecialCharacters)
+        .eq(0)
+        .should('be.visible');
+
+      cy.findAllByLabelText(labelStrong).should('be.visible');
+    });
+
+    it('changes the password minimum length value when "45" is typed in the input', () => {
```
Comment (bdauria) [inline centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:241]:
```suggestion
    it('updates the password minimum length value when the corresponding input is changed', () => {
```

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+      cy.findByLabelText(
+        `${labelMinimumTimeBetweenPasswordChanges} ${labelHour}`
+      ).type('{selectall}{backspace}');
+
+      cy.findByText(labelChooseADurationBetween1HourAnd1Week).should(
+        'not.exist'
+      );
+    });
+
+    it('selects the "Can reuse passwords" field when clicking on the corresponding switch', () => {
```
Comment (bdauria) [inline centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:419]:
```suggestion
    it('selects the "Can reuse passwords" field when the corresponding switch is clicked', () => {
```

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+
+const setComponentBeforeEach = (): void => {
+  const defaultPasswordSecurityPolicyURL = authenticationProvidersEndpoint(
+    Provider.Local
+  );
+
+  cy.interceptAPIRequest<PasswordSecurityPolicyToAPI>({
+    alias: 'getDefaultPasswordSecurityPolicyFromAPI',
+    method: Method.GET,
+    path: replace('./', '**', defaultPasswordSecurityPolicyURL),
```
Comment (Thebarda) [inline centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:79]: As we are starting using this function on our tests, I recommend to factorize it

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+    response: defaultPasswordSecurityPolicyAPI
+  });
+
+  cy.mount({
+    Component: (
+      <Router>
+        <div style={{ backgroundColor: '#fff' }}>
```
Comment (Thebarda) [inline centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:86]: We must move it into the custom command

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,603 @@
...
+        </div>
+      </Router>
+    )
+  });
+
+  cy.viewport(1200, 1000);
+};
+
+before(() => {
+  document.getElementsByTagName('body')[0].style = 'margin:0px';
```
Comment (Thebarda) [inline centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:97]: We must move it into the `mount` custom command

**Review by Thebarda — APPROVED** [review]

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx (hunk 4 above)
Comment (SoldierBoy31) [reply centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:79]: Are you talking about creating a custom command ?

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx (hunk 5 above)
Comment (SoldierBoy31) [reply centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:86]: This involves modifying all the tests that already call on this command, should I integrate it into this PR ? 
Also, some tests seems to call mount command differently (with Provider or not and Router or not), maybe we must to create another mount command for limite that  tests fail ?

File: centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx (hunk 6 above)
Comment (SoldierBoy31) [reply centreon/www/front_src/src/Authentication/Local/LocalAuthentication.cypress.spec.tsx:97]: This involves modifying all the tests that already call on this command, should I integrate it into this PR ?

**Review by Thebarda — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**